### PR TITLE
Feature: auth module

### PR DESCRIPTION
**Pull request by kingcody**
Changes:
* `injector:scripts` sorts `*.module.js` first 
* create auth module
* move all auth related config and functionality to auth module
* use named functions
* implement route authentication via router.decorator
* add `Auth.hasRole`
* implement `Util` service
* restrict `Authorization` header to same origin

### CODE REVIEW

File: app/templates/client/app/main/main.controller(js).js [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
+
+    $scope.$on('$destroy', function() {
+      socket.unsyncUpdates('thing');
+    });<% } %>
+  }
+
+  angular.module('<%= scriptAppName %>')
+    .controller('MainController', MainController);
 
 })();
```
Comment (Awk34) [inline app/templates/client/app/main/main.controller(js).js]: We should keep this tabbed the way it was. The IIFE shouldn't mess with our code's formatting.

Comment (kingcody) [inline app/templates/client/app/main/main.controller(js).js]: Sure...

**Comment by kingcody** [comment]
Closes #423, #790, #847, #1285

**Comment by Awk34** [comment]
Four issues in one PR? You should get a medal.

**Pull request merged**